ERCS/erc-7811.md

```diff
@@ -0,0 +1,239 @@
+
+Wallets already have information about a user's assets, including those not visible on-chain, and need a way to communicate that information to Dapps.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Method: `wallet_getAssets`
+
+#### Request schema
+
+```ts
+type Hex = `0x${string}`;
+
+type WalletGetAssetsRequest = {
+  account: Hex;
+  assetFilter?: Record<Hex, string[]>;
+  assetTypeFilter?: string[];
+  chainFilter?: Hex[];
+};
+```
+
+`account` is a **REQUIRED** field that indicates for which account assets are requested.
+
+If the `assetFilter` field is provided, the wallet **MUST** only return the assets specified within it, effectively disregarding the `assetTypeFilter` and `chainFilter` fields as they are already implicitly defined within the `assetFilter`.
+
+If the `assetFilter` field is omitted, the wallet **SHOULD** return all available assets for the requested account. It is **RECOMMENDED** that the returned assets be ordered by estimated value in descending order, as determined by the wallet.
+
+`assetTypeFilter` is an **OPTIONAL** field that specifies an array of asset types, as defined in this ERC. If `assetTypeFilter` field is provided, wallet **MUST** include only those assets in the response.
+
+`chainFilter` is an **OPTIONAL** field that specifies an array of chain ids, where each value in the array **MUST** be a valid [EIP-155](./eip-155.md) chainId
+
+#### Example request
+
+```json
+{
+  "account": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
+  "assetFilter": {
+    "0x1": ["0x456", "native"],
+    "0xa": ["0x789", "native"]
+  },
+  "assetTypeFilter": ["ERC20", "native"],
+  "chainFilter": ["0x1"]
+}
+```
+
+#### Response schema
+
+```ts
+type Asset = {
+  address: Hex | "native";
+  balance: Hex;
+  type: string;
+  metadata: any;
+};
+type WalletGetAssetsResponse = Record<Hex, Asset[]>;
+```
+
+The key **SHOULD** be [EIP-155](./eip-155.md) chainId
+
+Asset fields:
+
+`address` is the address of the asset as `Hex` or `native` string for native assets.
+
+`balance` is the balance of the asset as `Hex`
+
+**`type`:** A string indicating the type of the asset. Common asset types include but **aren’t limited to**:
+
+- **`ERC20`:** For [ERC-20](./eip-20.md) tokens
+- **`ERC721`:** For [ERC-721](./eip-721.md) tokens (NFTs)
+- **`native`:** For the chain's native asset
+
+**`metadata`:** An **OPTIONAL** object containing additional information about the asset. The specific fields within the metadata object may vary depending on the asset type and the wallet's implementation.
+
+#### Example response
+
+```json
+{
+  "0x1": [
+    {
+      "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+      "balance": "0xcaaea35047fe5702",
+      "type": "ERC20",
+      "metadata": {
+        "name": "Token",
+        "symbol": "TOK",
```

Comment:
what if wallet won't provide token TOK in the response? should the dapp fallback with it's own on chain discovery?

Reply:
It's a required field, so the wallet _must_ provide "TOK" in the response